variants/Generic_F411Rx/PeripheralPins.c

```diff
@@ -0,0 +1,259 @@
+/*
+ *******************************************************************************
+ * Copyright (c) 2019, STMicroelectronics
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ * 3. Neither the name of STMicroelectronics nor the names of its contributors
+ *    may be used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *******************************************************************************
+ * Automatically generated from STM32F411R(D-E)Tx.xml
```

Comment:
Where did you found this file ?
There is no `STM32F411R(D-E)Tx.xml`, it should be `STM32F411R(C-E)Tx.xml`
https://github.com/stm32duino/Arduino_Tools/tree/master/src/genpinmap/Arduino/STM32F411R(C-E)Tx

It seems you update manually the F401 one ? Right ?
Please use the one with above link (it misses 2 I2C pins in your one).

Reply:
Yes, I used the F401 as base. But I can't copy the contents from the PeripheralPins.c you linked in directly? Some lines have to be commented out like done on other pinouts?

Reply:
Yes you should comments some lines, it is user choice to select his peripheral pins mappings and this is the reason why all arrays are WEAKed to be able to redefine them as you want.
As the first pin found in the selected array is used then having multiple one is useless, so comment them out allow to save space.